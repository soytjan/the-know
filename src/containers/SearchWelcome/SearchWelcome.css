@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.SearchWel {
  background-color: #FBFCFC;
  border: 2px solid #979797;
  padding: 50px 70px;
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -40px 0 0 -320px; }
  .SearchWel h2 {
    font-size: 1.8em;
    padding-bottom: 20px;
    text-transform: uppercase; }
  .SearchWel h3,
  .SearchWel p {
    color: #979797; }
  .SearchWel h3 {
    font-size: 1.3em; }
  .SearchWel p {
    padding-bottom: 30px; }
  .SearchWel button {
    font-size: 1.2em;
    padding: 5px 25px;
    background-color: #BEECE9; }
    .SearchWel button:hover {
      cursor: pointer;
      background-color: #F8BFCD; }

.input-wel {
  font-size: 1.2em;
  width: 370px;
  height: 42px;
  padding: 5px; }

.current-btn {
  margin: 20px 0 30px; }

.btn-wel-search {
  height: 42px; }
  .btn-wel-search:hover {
    cursor: pointer; }
