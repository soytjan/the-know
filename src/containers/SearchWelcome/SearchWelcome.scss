@import "../../variables.scss";

.SearchWel {
  background-color: $white;
  border: 2px solid $gray;
  padding: 50px 70px;
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -140px 0 0 -320px;

  h2 {
    font-size: 1.8em;
    padding-bottom: 20px;
    text-transform: uppercase;
  }

  h3, 
  p {
    color: $gray;
  }

  h3 {
    font-size: 1.3em;
  }

  p {
    padding-bottom: 30px;
  }

  button {
    font-size: 1.2em;
    padding: 5px 25px;
    
    &:hover {
      background-color: $btn-hover-color;
    }
  }
}

.input-wel {
  font-size: 1.2em;
  width: 370px;
  height: 42px;
  padding: 5px;
}

.current-btn {
  margin: 20px 0 30px;
}

.btn-wel-search {
  height: 42px;

  &:hover {
    cursor: pointer;
  }
}

