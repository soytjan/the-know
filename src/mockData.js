export const mockEventData = {
  "last_item": null,
  "total_items": "9621",
  "first_item": null,
  "page_number": "1",
  "page_size": "10",
  "page_items": null,
  "search_time": "0.049",
  "page_count": "963",
  "events": {
    "event": [
      {
        "watching_count": null,
        "olson_path": "America/Denver",
        "calendar_count": null,
        "comment_count": null,
        "region_abbr": "CO",
        "postal_code": null,
        "going_count": null,
        "all_day": "0",
        "latitude": "39.7392358",
        "groups": null,
        "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
        "id": "E0-001-111209195-8",
        "privacy": "1",
        "city_name": "Denver",
        "link_count": null,
        "longitude": "-104.990251",
        "country_name": "United States",
        "country_abbr": "USA",
        "region_name": "Colorado",
        "start_time": "2018-07-05 09:00:00",
        "tz_id": null,
        "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
        "modified": "2018-02-19 06:12:10",
        "venue_display": "0",
        "tz_country": null,
        "performers": null,
        "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
        "venue_address": "Denver, Colorado, United States",
        "geocode_type": "EVDB Geocoder",
        "tz_olson_path": null,
        "recur_string": null,
        "calendars": null,
        "owner": "evdb",
        "going": null,
        "country_abbr2": "US",
        "image": null,
        "created": "2018-02-01 04:56:32",
        "venue_id": "V0-001-000179395-7",
        "tz_city": null,
        "stop_time": null,
        "venue_name": "Denver, Colorado, United States",
        "venue_url": "http://denver.eventful.com/venues/denver-colorado-united-states-/V0-001-000179395-7?utm_source=apis&utm_medium=apim&utm_campaign=apic"
      }
    ]
  }
}

export const mockCleanEventData = [
  {
    title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
    description: " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
    category: "event",
    venueName: "Denver, Colorado, United States", 
    region: "CO",
    postalCode: null,
    city: "Denver",
    startTime: "2018-07-05 09:00:00",
    venueAddress: "Denver, Colorado, United States",
    image: null,
    url: "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
    id: "E0-001-111209195-8"
  }
]