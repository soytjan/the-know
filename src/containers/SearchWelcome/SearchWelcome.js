import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchAndCleanGeocodeLocation, fetchAndCleanReverseGeocodeLocation } from '../../helper';
import { addEvents, addLocation } from '../../actions/';
import './SearchWelcome.css';

export class SearchWelcome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: ''
    };
  }

  handleChange = (event) => {
    const { value, name } = event.target;

    this.setState({ [name]: value });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { location } = this.state;
    const { addLocation, onReroute } = this.props;
    const geocodeLocation = await fetchAndCleanGeocodeLocation(location);
    addLocation(geocodeLocation);
    onReroute();
  }

  handleCurrentLocation = async () => {
    const { currentLocation, addLocation, onReroute } = this.props;
    const updatedLocation = await fetchAndCleanReverseGeocodeLocation(currentLocation.coordinates)
    
    addLocation(updatedLocation);
    onReroute();  
  }

  render() {
    return (
      <article className='SearchWel'>
        <h2>What's going on in...</h2>
        <h3>(Where ya at?)</h3>
        <button 
          className='current-btn'
          onClick={this.handleCurrentLocation}
        >
          USE MY CURRENT LOCATION
        </button>
        <p>- OR -</p>
        <form onSubmit={this.handleSubmit}>
          <input
            onChange={this.handleChange}
            value={this.state.location}
            name='location' 
            type="text"
            placeholder='Enter your city or location'
            className='input-wel'
          />
          <button className='btn-wel-search'>SEARCH</button>
        </form>
      </article>
    );
  }
}

SearchWelcome.propTypes = {
  currentLocation: PropTypes.object,
  addEvents: PropTypes.func,
  addLocation: PropTypes.func,
  onReroute: PropTypes.func
};

export const mapStateToProps = (state) => ({
  currentLocation: state.currentLocation
});

export const mapDispatchToProps = (dispatch) => ({
  addEvents: events => dispatch(addEvents(events)),
  addLocation: location => dispatch(addLocation(location))
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchWelcome);