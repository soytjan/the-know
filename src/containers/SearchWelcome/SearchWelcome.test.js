import React from 'react';
import { shallow } from 'enzyme';
import { mockGeocodeData, mockCleanGeocodeData } from '../../mockData';
import { SearchWelcome } from './SearchWelcome';

describe('SearchWelcome', () => {
  let renderedComponent;
  let mockAddEvents;
  let mockAddLocation;
  let mockOnReroute;
  let mockCurrentLocation;
  
  beforeAll(() => {
    mockCurrentLocation = {
      coordinates: {
        "lat": 39.7380371,
        "lng": -105.02651949999999
      }
    }
  })

  beforeEach(() => {
    mockAddEvents = jest.fn();
    mockAddLocation = jest.fn();
    mockOnReroute = jest.fn();
    renderedComponent = shallow(
      <SearchWelcome
        currentLocation={mockCurrentLocation} 
        addEvents={mockAddEvents}
        addLocation={mockAddLocation}
        onReroute={mockOnReroute}
      />
    );
  })

  it.skip('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should update state on input change', () => {
    const mockEvent = { target: { value: 'abc', name: 'location' }};
    const expected = 'abc';

    renderedComponent.instance().handleChange(mockEvent);

    renderedComponent.update();

    expect(renderedComponent.state().location).toEqual(expected);
  });

  it('should fetch and clean coordinates for location on handleSubmit', () => {
    
  });

  it('should fetch and clean upcoming events for location on submit', () => {

  });

  it('should call addEvents to store on submit', () => {

  });

  it('should call addLocation to store on submit', () => {

  });

  it('should set location in local storage on submit', () => {

  });

  it('should call onReroute on submit', () => {

  });

  it('should fetch and clean event data based on currentLocation', () => {

  });

})