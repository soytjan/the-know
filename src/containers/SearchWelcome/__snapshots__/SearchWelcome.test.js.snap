// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchWelcome should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchWelcome
    addEvents={[Function]}
    addLocation={[Function]}
    currentLocation={
        Object {
            "coordinates": Object {
              "lat": 39.7380371,
              "lng": -105.02651949999999,
            },
          }
    }
    onReroute={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          What's going on in...
</h2>,
        <h3>
          (Where ya at?)
</h3>,
        <button
          className="current-btn"
          onClick={[Function]}
>
          USE MY CURRENT LOCATION
</button>,
        <p>
          - OR -
</p>,
        <form
          onSubmit={[Function]}
>
          <input
                    className="input-wel"
                    name="location"
                    onChange={[Function]}
                    placeholder="Enter your city or location"
                    type="text"
                    value=""
          />
          <button
                    className="btn-wel-search"
          >
                    SEARCH
          </button>
</form>,
      ],
      "className": "SearchWel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "What's going on in...",
        },
        "ref": null,
        "rendered": "What's going on in...",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "(Where ya at?)",
        },
        "ref": null,
        "rendered": "(Where ya at?)",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "USE MY CURRENT LOCATION",
          "className": "current-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "USE MY CURRENT LOCATION",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "- OR -",
        },
        "ref": null,
        "rendered": "- OR -",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input-wel"
              name="location"
              onChange={[Function]}
              placeholder="Enter your city or location"
              type="text"
              value=""
/>,
            <button
              className="btn-wel-search"
>
              SEARCH
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-wel",
              "name": "location",
              "onChange": [Function],
              "placeholder": "Enter your city or location",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SEARCH",
              "className": "btn-wel-search",
            },
            "ref": null,
            "rendered": "SEARCH",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            What's going on in...
</h2>,
          <h3>
            (Where ya at?)
</h3>,
          <button
            className="current-btn"
            onClick={[Function]}
>
            USE MY CURRENT LOCATION
</button>,
          <p>
            - OR -
</p>,
          <form
            onSubmit={[Function]}
>
            <input
                        className="input-wel"
                        name="location"
                        onChange={[Function]}
                        placeholder="Enter your city or location"
                        type="text"
                        value=""
            />
            <button
                        className="btn-wel-search"
            >
                        SEARCH
            </button>
</form>,
        ],
        "className": "SearchWel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "What's going on in...",
          },
          "ref": null,
          "rendered": "What's going on in...",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "(Where ya at?)",
          },
          "ref": null,
          "rendered": "(Where ya at?)",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "USE MY CURRENT LOCATION",
            "className": "current-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "USE MY CURRENT LOCATION",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "- OR -",
          },
          "ref": null,
          "rendered": "- OR -",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="input-wel"
                name="location"
                onChange={[Function]}
                placeholder="Enter your city or location"
                type="text"
                value=""
/>,
              <button
                className="btn-wel-search"
>
                SEARCH
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-wel",
                "name": "location",
                "onChange": [Function],
                "placeholder": "Enter your city or location",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SEARCH",
                "className": "btn-wel-search",
              },
              "ref": null,
              "rendered": "SEARCH",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
